### PR TITLE
[MB-15593] Refactor `SubmitSITExtensionsModal.jsx` component

### PR DESCRIPTION
**Pull request by afoon**
## Summary

An additional backend ticket will make the service changes need to calculate the change in the number of days.

- [ ]  Update UI in SubmitSITExtensionsModal
- [ ]  Update test for SubmitSITExtensionsModal
- [ ]  Update SubmitSITExtensionsModal story

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the Storybook locally.

```sh
make storybook
```

##### Terminal 2

Start the UI locally.

```sh
make client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the Office App
2. Login as a TOO
3. Choose the  move code with `SITEXT`
4. Go to the Move Task Order tab
5. Click the edit button on the SIT display
     - [ ] Submit modal UI looks like Figma spec
6. Edit the Total Days Approved
    - [ ] The Sit authorized date changes accordingly
7. Edit the Sit End date
    - [ ] The Total days approved changes accordingly

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/main/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

## Screenshots
Screenshot 2023-04-20 at 1 17 39 AM

image

### CODE REVIEW

File: src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx [hunk 1]
```diff
@@ -1,29 +1,140 @@
...
 import classnames from 'classnames';
-import { Formik, Field } from 'formik';
+import { Formik, Field, useField } from 'formik';
 import PropTypes from 'prop-types';
 import * as Yup from 'yup';
 import { Button, Label, Textarea } from '@trussworks/react-uswds';
+import moment from 'moment';
+
+import DataTableWrapper from '../../DataTableWrapper/index';
+import DataTable from '../../DataTable/index';
```
Comment (YanZ777) [inline src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx:10]: **suggestion:** Can we switch these to absolute imports? 
```suggestion
import DataTableWrapper from 'components/DataTableWrapper';
import DataTable from 'components/DataTable';
```

File: swagger-def/prime.yaml [hunk 2]
```diff
@@ -1234,6 +1234,7 @@ definitions:
       requestedDays:
         type: integer
         example: 30
+        minimum: 1
```
Comment (rogeruiz) [inline swagger-def/prime.yaml:1237]: Good catch on this one @michaelvaldes-truss

**Review by rogeruiz — APPROVED** [review]
lgtm 🌈

Paired with @michaelvaldes-truss and @ruizajtruss to verify it is working.

**Review by michaelvaldes-truss — APPROVED** [review]
lgtm; both inputs update each other appropriately and behave as expected for shipments with one or two SIT service items

File: src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx [hunk 3]
```diff
@@ -1,29 +1,158 @@
...
+      .required('Required'),
+    sitEndDate: Yup.date().min(
+      moment(sitStatus.sitEntryDate).add(1, 'days').format('DD MMM YYYY'),
+      'The end date must occur after the start date. Please select a new date.',
+    ),
+  });
 
-const SubmitSITExtensionModal = ({ onClose, onSubmit, summarySITComponent }) => {
   return (
     <div>
```
Comment (YanZ777) [inline src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx:157]: **suggestion:** If you use `connectModal` from `src/components/Modal/Modal.jsx`, you won't need all this extra boilerplate. From there you'd be able to import the connected version of this modal.

**info:** `src/components/EulaModal/index.jsx` and `src/containers/Headers/LoggedOutHeader.jsx` models what to do.

File: src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx [hunk 4]
```diff
@@ -1,29 +1,158 @@
...
       <Overlay />
```
Comment (YanZ777) [inline src/components/Office/SubmitSITExtensionModal/SubmitSITExtensionModal.jsx:158]: **suggestion:** If you use `connectModal` from `src/components/Modal/Modal.jsx`, you won't need all this extra boilerplate. From there you'd be able to import the connected version of this modal.

**info:** `src/components/EulaModal/index.jsx` and `src/containers/Headers/LoggedOutHeader.jsx` models what to do.

**Review by AnnaGingle — APPROVED** [review]
LGTM

**Pull request merged**